### PR TITLE
docs(revert): fix broken links under user docs and obsolete info from publishing

### PR DESCRIPTION
**Pull request by mohanraj-r**
- Partially reverts changes from https://github.com/forcedotcom/salesforcedx-vscode/pull/3813/files that was based on misunderstanding of how the doc system works
  - Fixes couple of broken links reported in https://github.com/forcedotcom/salesforcedx-vscode/issues/3828
- Adds clarifying info to the docs about how relative URLs work

### What issues does this PR fix or reference?
@W-10869967@

### CODE REVIEW

**Comment by mohanraj-r** [comment]
Thanks @randi274 
There is doc for running the doc site locally https://github.com/forcedotcom/salesforcedx-vscode/blob/mohanraj-r/doc_fix_misc/docs/README.md#how-to-run-the-doc-site-locally
I haven't done it yet 😞  - It has been on my TODO list since its a diff tech stack.

**Review by randi274 — APPROVED** [review]
Thanks @mohanraj-r - I was struggling to test it as well, and ran into a few different errors. Mostly a ruby version mismatch due to what I had locally vs tried to install with homebrew. I wasn't able to resolve them, but comparing it with what you're reverting, if it worked before, it should work again!

File: docs/_articles/en/troubleshooting.md [hunk 1]
```diff
@@ -14,23 +14,23 @@ If you don’t see any SFDX commands in the command palette, make sure that you
...
-See: [Java Configuration](./getting-started/java-setup.md)
+See: [Java Configuration](./en/getting-started/java-setup)
 
 ## Set Salesforce CLI Path (Windows)
 
 See [Installing the Salesforce CLI](https://developer.salesforce.com/tools/sfdxcli). After installing Salesforce CLI, if you get the error 'Salesforce CLI is not installed', most likely Salesforce CLI is not added as a path variable. To verify or add Salesforce CLI to your Windows path variable:
 
 1. Search for `Edit the system environment variables`.
-   <img align="center" width="80%" height="80%" src="../../images/troubleshooting_set_cli_path.1.png" alt="Windows Search">
+   <img align="center" width="80%" height="80%" src="./images/troubleshooting_set_cli_path.1.png" alt="Windows Search">
```
Comment (gbockus-sf) [inline docs/_articles/en/troubleshooting.md:24]: Screen Shot 2022-03-22 at 12 01 29 PM

File: docs/_articles/en/troubleshooting.md [hunk 2]
```diff
@@ -14,23 +14,23 @@ If you don’t see any SFDX commands in the command palette, make sure that you
...
 
 1. In `System Properties`, click `Environment Variables`.
-   <img align="center" width="80%" height="80%" src="../../images/troubleshooting_set_cli_path.2.png" alt="System Properties">
+   <img align="center" width="80%" height="80%" src="./images/troubleshooting_set_cli_path.2.png" alt="System Properties">
 
 1. Under `User Variables`, double-click `Path` .
-   <img align="center" width="80%" height="80%" src="../../images/troubleshooting_set_cli_path.3.png" alt="User Variables">
+   <img align="center" width="80%" height="80%" src="./images/troubleshooting_set_cli_path.3.png" alt="User Variables">
```
Comment (gbockus-sf) [inline docs/_articles/en/troubleshooting.md:31]: Screen Shot 2022-03-22 at 12 02 49 PM

File: docs/_articles/en/troubleshooting.md [hunk 3]
```diff
@@ -14,23 +14,23 @@ If you don’t see any SFDX commands in the command palette, make sure that you
...
 
 1. Verify Salesforce CLI is listed as an entry. Otherwise, click New and paste the path where you installed CLI. For example, `C:\Program Files\Salesforce CLI\bin`
-   <img align="center" width="80%" height="80%" src="../../images/troubleshooting_set_cli_path.4.png" alt="Path View">
+   <img align="center" width="80%" height="80%" src="./images/troubleshooting_set_cli_path.4.png" alt="Path View">
```
Comment (gbockus-sf) [inline docs/_articles/en/troubleshooting.md:34]: Screen Shot 2022-03-22 at 12 02 07 PM

File: docs/_articles/en/troubleshooting.md [hunk 4]
```diff
@@ -60,7 +60,7 @@ If your deployment times for Apex code are slow, your org might have the `Compil
 
 ## Other Resources
 
-For Apex Debugger troubleshooting information, see [Apex Interactive Debugger](./apex/interactive-debugger.md).
+For Apex Debugger troubleshooting information, see [Apex Interactive Debugger](./en/apex/interactive-debugger).
```
Comment (gbockus-sf) [inline docs/_articles/en/troubleshooting.md:63]: verified link works locally.

File: docs/_articles/en/troubleshooting.md [hunk 5]
```diff
@@ -14,23 +14,23 @@ If you don’t see any SFDX commands in the command palette, make sure that you
 
 ## Set Your Java Version
 
-See: [Java Configuration](./getting-started/java-setup.md)
+See: [Java Configuration](./en/getting-started/java-setup)
```
Comment (gbockus-sf) [inline docs/_articles/en/troubleshooting.md:17]: verified link works locally for en and ja

**Comment by mohanraj-r** [comment]
Fixes couple of broken links reported in #3828

**Comment by mohanraj-r** [comment]
@randi274 @gbockus-sf discovered that we can preview the rendered doc site from a Github Pull request by clicking "Details" against the "netlify/salesforcedx-vscode/deploy-preview — Deploy Preview ready!" check. Added a note to Readme.

File: docs/README.md [hunk 6]
```diff
@@ -1,6 +1,8 @@
 # How to run the doc site locally
 
-The `docs` folder contains content published to [Salesforce Extensions for Visual Studio Code](https://developer.salesforce.com/tools/vscode)
+- The `docs` folder contains content published to [Salesforce Extensions for Visual Studio Code](https://developer.salesforce.com/tools/vscode).
+- To preview the rendered doc site from a Github Pull request click "Details" against the "netlify/salesforcedx-vscode/deploy-preview — Deploy Preview ready!" check.
```
Comment (gbockus-sf) [inline docs/README.md:4]: where does this live?  I don't see that check. 
Screen Shot 2022-03-22 at 1 16 17 PM

File: docs/_articles/en/index.md [hunk 7]
```diff
@@ -6,7 +6,7 @@ permalink: /
 
 The Salesforce Extension pack includes tools for developing on the Salesforce platform in the lightweight, extensible VS Code editor. These tools provide features for working with development orgs (scratch orgs, sandboxes, and DE orgs), Apex, Aura components, and Visualforce.
 
-![GIF showing Apex code completion, pushing source to a scratch org, and running Apex tests](../../images/overview.gif)
+![GIF showing Apex code completion, pushing source to a scratch org, and running Apex tests](./images/overview.gif)
```
[the remark this answers is not in the document]
Comment (sbudhirajadoc) [reply docs/_articles/en/index.md:10]: I don't think this image path is correct.

File: docs/README.md (hunk 6 above)
Comment (mohanraj-r) [reply docs/README.md:4]: That is weird. I see it right in the center among the checks 

image

**Review by sbudhirajadoc — APPROVED** [review]

**Pull request merged**